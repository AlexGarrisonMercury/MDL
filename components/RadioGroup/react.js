import React, { memo } from 'react'
import PropTypes from 'prop-types'
import { React as Radio } from 'components/Radio'
import classNames from 'classnames/bind'
import s from './style.module.scss'

const cx = classNames.bind(s)

const RadioGroup = ({
  checkedChoice, choices, row, onChange,
}) => {
  const radios = choices.map(({ label, value }) => (
    <li key={value} className={cx(['radio-group__li'])}>
      <Radio
        checked={checkedChoice === value}
        onClick={onChange}
        onChange={onChange}
        value={value}
        label={label}
      />
    </li>
  ))
  return <ul className={cx(['radio-group', row && 'radio-group--row'])}>{radios}</ul>
}

RadioGroup.propTypes = {
  checkedChoice: PropTypes.string,
  row: PropTypes.bool,
  choices: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
      checked: PropTypes.bool,
    }),
  ).isRequired,
  onChange: PropTypes.func.isRequired,
}

RadioGroup.defaultProps = {
  checkedChoice: null,
  row: false,
}

export default memo(RadioGroup)
