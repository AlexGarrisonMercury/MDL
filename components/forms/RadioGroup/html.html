<ul class="radio-group">
  <!-- foreach -->
  <li class="radio-group__li">
    <!-- <Radio /> -->
  </li>
  <!-- endforeach -->
</ul>
