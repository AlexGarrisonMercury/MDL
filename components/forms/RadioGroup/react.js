import React, { memo } from 'react'
import PropTypes from 'prop-types'
import { React as Radio } from 'components/forms/RadioButton'
import s from './style.module.scss'

const RadioGroup = ({
  checkedChoice, choices, direction, onChange,
}) => {
  const radios = choices.map(({ label, value }) => (
    <li key={value} className={direction === 'column' ? '' : s['radio-group__row']}>
      <Radio
        checked={checkedChoice === value}
        onClick={event => onChange(event.target.value)}
        onChange={event => onChange(event.target.value)}
        label={label}
        value={value}
      />
    </li>
  ))
  return <ul className={s['radio-group']}>{radios}</ul>
}

RadioGroup.propTypes = {
  checkedChoice: PropTypes.string,
  direction: PropTypes.string,
  choices: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string,
      checked: PropTypes.bool,
    }),
  ).isRequired,
  onChange: PropTypes.func.isRequired,

}

RadioGroup.defaultProps = {
  checkedChoice: null,
  direction: 'column',
}

export default memo(RadioGroup)
