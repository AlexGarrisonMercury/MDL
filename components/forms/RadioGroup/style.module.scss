@import "styles/colors.scss";

ul {
  list-style-type: none;
}

.radio-group-row {
  display: inline-block;
  padding: 5px;
}
