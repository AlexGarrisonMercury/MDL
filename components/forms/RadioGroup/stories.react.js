import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'
import { withState } from '@dump247/storybook-state'
import Radio from './react'
import RadioGroup from './react'

storiesOf('Forms/RadioGroup', module)
  .addDecorator(withKnobs)
  .add(
    'checked with a click event',
    withState({
      checkedChoice: null,
      choices: [
        {
          label: 'Filter 1',
          value: 'FILTER1',
        },
        {
          label: 'Filter 2',
          value: 'FILTER2',
        },
      ],
      direction: 'column',
    })(({ store }) => (
      <React.Fragment>
        <RadioGroup
          checkedChoice={store.state.checkedChoice}
          choices={store.state.choices}
          onChange={value => store.set({ checkedChoice: value })}
          direction
        />
      </React.Fragment>
    )),
  )
