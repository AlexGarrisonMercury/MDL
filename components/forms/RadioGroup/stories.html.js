import { storiesOf } from '@storybook/html'
import html from './html.html'

storiesOf('Forms/RadioGroup', module).add('default', () => html)
