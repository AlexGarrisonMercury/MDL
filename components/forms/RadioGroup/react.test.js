import React from 'react'
import { mount } from 'enzyme'
import RadioGroup from './react'


describe('RadioButton', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = {
      checked: jest.fn(),
      onChange: jest.fn(),
      onClick: jest.fn(),
      choices: []
    }

    wrapper = mount(<RadioGroup {...props} />)
  })

  describe('The default state of the component with minimal required items passed in', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('Should contain no list items if the choices passed in are an empty array', () => {
      expect(wrapper.find('li').length).toBe(0)
    })
  })

  describe('The radio buttons should render with passed in props and using the default direction', () => {
    let listItems

    beforeEach(() => {
      props = {
        ...props,
        choices: [
          {
            label: 'Filter 1',
            value: 'FILTER1',
          },
          {
            label: 'Filter 2',
            value: 'FILTER2',
          },
        ]
      }
      wrapper = mount(<RadioGroup {...props} />)
      listItems = wrapper.find('li')
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('Should render li element equivalent to the number of choices passed in', () => {
      expect(listItems.length).toBe(props.choices.length)
    })

    it('Should render with no className on the list items if using the default direction', () => {
      wrapper.find('li').forEach( listItem => {
        expect(listItem.props().className).toBe('')
      })
    })

    it('Should render each list item with the key being the value passed into them from the choices', () => {
      wrapper.find('li').forEach( (listItem, index) => {
        expect(listItem.key()).toBe(props.choices[index].value)
      })
    })
  });

  describe('The radio buttons should render with passed in props and a direction set to anything other than "column"', () => {
    beforeEach(() => {
      props = {
        ...props,
        direction: '',
        choices: [
          {
            label: 'Filter 1',
            value: 'FILTER1',
          },
          {
            label: 'Filter 2',
            value: 'FILTER2',
          },
          {
            label: 'Filter 3',
            value: 'FILTER 3'
          }
        ]
      }
      wrapper = mount(<RadioGroup {...props} />)
    })

    it('Should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('Should render li element equivalent to the number of choices passed in', () => {
      expect(wrapper.find('li').length).toBe(props.choices.length)
    })

    it('Should render with no className on the list items if using the default direction', () => {
      wrapper.find('li').forEach(listItem => {
          expect(listItem.props().className).toBe('radio-group__row')
        })
    })
  })
})
