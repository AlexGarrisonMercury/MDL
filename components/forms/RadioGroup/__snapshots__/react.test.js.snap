// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup The radio buttons should render with passed in props and a direction set to anything other than "column" Should render correctly 1`] = `
<ul
  className="radio-group"
>
  <li
    className="radio-group__li"
    key="FILTER1"
  >
    <Radio
      checked={false}
      label="Filter 1"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER1"
    />
  </li>
  <li
    className="radio-group__li"
    key="FILTER2"
  >
    <Radio
      checked={false}
      label="Filter 2"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER2"
    />
  </li>
  <li
    className="radio-group__li"
    key="FILTER 3"
  >
    <Radio
      checked={false}
      label="Filter 3"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER 3"
    />
  </li>
</ul>
`;

exports[`RadioGroup rendering with defaults should render correctly 1`] = `
<ul
  className="radio-group"
/>
`;

exports[`RadioGroup rendering with props should render correctly 1`] = `
<ul
  className="radio-group"
>
  <li
    className="radio-group__li"
    key="FILTER1"
  >
    <Radio
      checked={false}
      label="Filter 1"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER1"
    />
  </li>
  <li
    className="radio-group__li"
    key="FILTER2"
  >
    <Radio
      checked={false}
      label="Filter 2"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER2"
    />
  </li>
</ul>
`;

exports[`RadioGroup rendering with props should render correctly with row passed in 1`] = `
<ul
  className="radio-group radio-group--row"
>
  <li
    className="radio-group__li"
    key="FILTER1"
  >
    <Radio
      checked={false}
      label="Filter 1"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER1"
    />
  </li>
  <li
    className="radio-group__li"
    key="FILTER2"
  >
    <Radio
      checked={false}
      label="Filter 2"
      onChange={[MockFunction]}
      onClick={[MockFunction]}
      value="FILTER2"
    />
  </li>
</ul>
`;
